### PR TITLE
Implement "add new" and "rename" for solution folder files

### PR DESCRIPTION
**Pull request by mhutch**
Fixes issue #4775 - Cannot add new file to solution items
Fixes issue #4774 - Cannot rename solution items

### CODE REVIEW

**Comment by p-lad** [comment]
Hello,
Following is the status :-

Build Taken from :- https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1688171&tab=artifacts&_a=summary

Observations :- 
i. When any item is added to solutions , we can now see the “Rename” option available and able to rename the solution items successfully.
ii. When we right click on the Solution , there is option "ADD NEW FILE" and on clicking we can add file to solution successfully.

Build Info :-
Visual Studio Enterprise 2017 for Mac (Preview) Version 7.6 Preview (7.6 build 808)
Mono 5.14.0.83 (2018-04/769481d834c) (64-bit)
Xamarin.Mac Version: 4.5.0.313 (Visual Studio Enterprise)
Xamarin.iOS Version: 11.11.0.313 (Visual Studio Enterprise)
Xamarin.Android Version: 8.3.99.12 (Visual Studio Enterprise)

have also marked following bugs as QA Approved :- 
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/616402
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/616403

Screenshot link :- http://recordit.co/XGQMbb1coU

Hence marking this as “QA Approved ”

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/SolutionNodeBuilder.cs [hunk 1]
```diff
@@ -333,16 +333,18 @@ public void OnEditSolutionUpdate (CommandInfo info)
 			var solution = (Solution) CurrentNode.DataItem;
 			info.Visible = info.Enabled = !string.IsNullOrEmpty (solution.FileName) && File.Exists (solution.FileName);
 		}
-		
-/*		[CommandHandler (ProjectCommands.AddNewFiles)]
-		protected void OnAddNewFiles ()
+
+		[CommandHandler (ProjectCommands.AddNewFiles)]
+		protected async void OnAddNewFiles ()
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/SolutionNodeBuilder.cs]: Why does this need to be async?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/SolutionFolderNodeBuilder.cs [hunk 2]
```diff
@@ -321,16 +321,18 @@ public void OnUpdateReload (CommandInfo info)
 			}
 		}
 		
-/*		[CommandHandler (ProjectCommands.AddNewFiles)]
-		protected void OnAddNewFiles ()
+		[CommandHandler (ProjectCommands.AddNewFiles)]
+		protected async void OnAddNewFiles ()
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/SolutionFolderNodeBuilder.cs]: Why does this need to be async?

Comment (mhutch) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/SolutionFolderNodeBuilder.cs]: So if there's an exception in the called async method, we get a callstack with caller info

**Pull request merged**